### PR TITLE
changed metadata import to allow the groups arg

### PR DESCRIPTION
**Pull request by jspaezp**
solves: https://github.com/wfondrie/mokapot/issues/85
Also adds multiple python versions to the test matrix in the github worflows

### CODE REVIEW

File: .github/workflows/tests.yml
Comment (wfondrie) [inline .github/workflows/tests.yml:22]: **suggestion:** I learn toward just removing 3.11 for now.

File: mokapot/plugins/__init__.py
Comment (wfondrie) [inline mokapot/plugins/__init__.py:8]: **todo:** Please add a comment suggesting we switch to `importlib.metadata` if/when our min version is 3.10.

**Review by wfondrie — APPROVED** [review]
Nice work @jspaezp - this looks good to me. 

Please just add the comment I suggested and update the changelog before we merge it!

File: mokapot/model.py
Comment (jspaezp) [reply mokapot/model.py:160]: I do not believe we need it ... since the BaseSearchCV class has an estimator attr ...

File: .github/workflows/tests.yml
Comment (jspaezp) [reply .github/workflows/tests.yml:22]: OK, we can do that (nontheless, check shows as passing in the summary view https://github.com/wfondrie/mokapot/actions). LMK if this is the last call

File: mokapot/plugins/__init__.py
Comment (jspaezp) [reply mokapot/plugins/__init__.py:8]: right on!

**Comment by wfondrie** [comment]
Perfect! I'll merge once CI has completed successfully 🙌

File: .github/workflows/tests.yml
Comment (wfondrie) [reply .github/workflows/tests.yml]: Yes, at least for now.

**Pull request merged**
